ReadersAndroidJetpack/app/src/main/java/com/kontinua/readersandroidjetpack/views/PDFView.kt

```diff
@@ -124,5 +137,20 @@ fun PDFViewer(
                 pan = panOffset.value
             )
         }
+        IconButton(
+            onClick = {
+                workbook?.id?.let { wbId ->
+                    bookmarkViewModel.toggleBookmark(wbId, navbarManager.pageNumber)
+                }
+            },
+            modifier = Modifier
+                .align(Alignment.TopEnd)
+                .padding(16.dp)
+        ) {
+            Icon(
+                imageVector = if (isPageBookmarked) Icons.Filled.Bookmark else Icons.Filled.BookmarkBorder,
+                contentDescription = if (isPageBookmarked) "Remove Bookmark" else "Add Bookmark"
```

Comment:
Move presentation strings to strings.xml

I’d also check how similar bookmark functionality is often surfaced in semantics. It seems more like a toggle or checkmark with a selected state. Currently that “is bookmarked” state is latent in the change in action, which doesn’t really communicate what the visual icon un/filled state says.

Reply:
Ah here’s an example https://developer.android.com/develop/ui/compose/accessibility/key-steps#describe_an_elements_state